### PR TITLE
BDN11 - Added backwards compatible frameworks and changes

### PR DESCRIPTION
**Pull request by glitch100**
## Overview
This branch includes changes to support older frameworks as various users have expressed that they are still on .net 4.5.1~.

This resolves issue #11 and supercedes the following PR https://github.com/glitch100/BinanceDotNet/pull/9  thanks @robbaman 

### Pre-steps
To install please use the following NuGet. Note that it is a pre-release until such time the `NLog` abstraction had been refactored.

[https://www.nuget.org/packages/BinanceDotNet/4.0.0-rc1](https://www.nuget.org/packages/BinanceDotNet/4.0.0-rc1)

`Install-Package BinanceDotNet -Version 4.0.0-rc1 -Prerelease` 
At this point it is critical to include the `-Prerelease` tag

### Confirmations 
- [x] Can be installed on .NET 4.5.1 
- [x] Can be installed on .NET 4.5.2 
- [x] Can be installed on dotnetcore

## To-do
- [x] ~Use logging abstractions and remove dependency on `NLog.Extensions.Logging`~
      - Removed `NLog` and replaced with `log4net` until such time abstractions can be brought in

### CODE REVIEW

File: BinanceExchange.API/RequestClient.cs [hunk 1]
```diff
@@ -194,9 +196,15 @@ public static void SetAPIKey(string key)
         /// <param name="signatureRawData"></param>
         /// <param name="receiveWindow"></param>
         /// <returns></returns>
+        /// 
         private static Uri CreateValidUri(Uri endpoint, string secretKey, string signatureRawData, long receiveWindow)
         {
-            var timestamp = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds().ToString();
+            string timestamp;
+#if NETSTANDARDFEATURE
```
Comment (robbaman) [inline BinanceExchange.API/RequestClient.cs]: NETSTANDARDFEATURE is not a recognized moniker, it should be NETSTANDARD2_0

File: BinanceExchange.API/Websockets/AbstractBinanceWebSocketClient.cs [hunk 2]
```diff
@@ -147,7 +147,14 @@ private Guid CreateUserDataBinanceWebSocket(Uri endpoint, UserDataWebSocketMessa
                     }
                 };
             };
+#if NETSTANDARDFEATURE
```
Comment (robbaman) [inline BinanceExchange.API/Websockets/AbstractBinanceWebSocketClient.cs]: NETSTANDARDFEATURE is not a recognized moniker, it should be NETSTANDARD2_0

File: BinanceExchange.API/Websockets/AbstractBinanceWebSocketClient.cs [hunk 3]
```diff
@@ -181,7 +188,14 @@ private Guid CreateUserDataBinanceWebSocket(Uri endpoint, UserDataWebSocketMessa
...
```
Comment (robbaman) [inline BinanceExchange.API/Websockets/AbstractBinanceWebSocketClient.cs]: NETSTANDARDFEATURE is not a recognized moniker, it should be NETSTANDARD2_0, also note that the TryAdd() method on a dictionary is only available when targetting NETCOREAPP2_0. So in all targetframeworks this solution uses the ContainsKey() approach is required.

**Comment by robbaman** [comment]
Hey Jon, it's good to see you're making good progress. I apologize for my lack of contributions, contrary to implying to contribute more, I've unfortunately been swamped with work past week.

Just some quick input though, I noticed you're started introducing the Abstractions project, that's great. Do note however that in the standard Microsoft has set out, the namespace of the items in the Abstractions package does not actually include the *Abstractions* section.

For instance the `ISymbolWebSocketResponse` interface should not be in the *BinanceExchange.Abstractions.Models.WebSocket.Interfaces* namespace, but instead the *BinanceExchange.Models.WebSocket.Interfaces* namespace.

See for instance this class in the Microsoft.AspNetCore.Mvc.Abstractions package https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Abstractions/Filters/ExceptionContext.cs

Also with regard to the logging, did you change NLog to log4net? Does this mean you've abandoned the idea of using the generic Microsoft Logging abstractions or is this merely a test? 

I did actually attempt to incorporate the ILogger abstractions into the project but ran into the issue that the current code regularly creates loggers manually where the recommended approach is to have them constructor injected. This of course is a challenge until such a time that the project is converted to a more IoC friendly structure (i.e. no static classes)

**Comment by glitch100** [comment]
Hey @robbaman ,

It's ok I have also been swamped at work so finding it hard to get lots done in one go. I see what you are saying, not an actual `Abstractions` project? That's cool. We can move that back the way it should be 👍 

As for the `NLog` -> `log4net`. The first issue that although `NLog` was a good choice all the packages are prerelease and as such create issues unless I keep my package as prerelease. I attempted to bring in the MS Logging Abstractions (https://www.nuget.org/packages/Microsoft.Extensions.Logging.Abstractions) however it causes compatibility issues in .NET 4.5.1~. Which is incredibly annoying as it would solve my problem perfectly. As such for now using log4net enables full compatibility across all versions and would also allow users to define their own appenders etc, until such time the above package gets the support.

That said if you are aware of a way to use it that would be great.

As for standard of using Loggers in an IoC manner I would be down for that but until such time we can get the Abstractions in, passing in loggers and letting them be setup at a class level isn't that bad an approach.

**Pull request merged**